File: schemas/podcast-platform/drizzle/chapters.ts
// Chapter markers within an episode for navigation and structure
// See README.md for full design rationale.

import { pgTable, uuid, text, boolean, integer, timestamp, index } from "drizzle-orm/pg-core";
import { episodes } from "./episodes";

export const chapters = pgTable(
  "chapters",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    episodeId: uuid("episode_id").notNull().references(() => episodes.id, { onDelete: "cascade" }),
    startTimeMs: integer("start_time_ms").notNull(),
    endTimeMs: integer("end_time_ms"),
    title: text("title").notNull(),
    url: text("url"),
    imageUrl: text("image_url"),
    isHidden: boolean("is_hidden").notNull().default(false),
    position: integer("position").notNull(),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_chapters_episode_id_start_time_ms").on(table.episodeId, table.startTimeMs),
  ]
);

File: schemas/podcast-platform/spacetimedb/chapters.rs
// chapters: Timestamped chapter markers within an episode.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = chapters, public)]
pub struct Chapter {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub episode_id: String, // UUID — FK → episodes.id (cascade delete)
    pub start_time_ms: i32,
    pub end_time_ms: Option<i32>,
    pub title: String,
    pub url: Option<String>,
    pub image_url: Option<String>,
    pub is_hidden: bool,
    pub position: i32,
    pub created_at: Timestamp,
}
// Composite index: (episode_id, start_time_ms) — not supported, enforce in reducer logic
